test/assertGuards.ts

```diff
@@ -14,9 +14,9 @@ function assertGuardConfirmed<FROM, TO extends FROM>(
 	assert.isAtLeast(cs.length, 1, 'no confirmation reason for successful guard');
 }
 
-function assertGuardFailed<FROM, TO extends FROM>(
+function assertGuardFailed<FROM, MID extends FROM, TO extends FROM>(
```

Comment:
I don't think this change is necessary.

Reply:
typescript rejects my test data for the object guards without it, because some of the test data for the object guards are "half way" between `FROM` and `TO`

Reply:
That seems off. If MID is in the middle, then MID extends FROM and you can pass a MID value in a FROM parameter. Would have to dig more to see what's up...